aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -105,6 +105,30 @@ static void cumprod_cpu_kernel(Tensor& result, const Tensor& self, int64_t dim)
   });
 }
 
+static void logcumsumexp_cpu_kernel(Tensor& result, const Tensor& self, int64_t dim) {
+  auto wrap_dim = maybe_wrap_dim(dim, self.dim());
+  int64_t self_dim_size = ensure_nonempty_size(self, wrap_dim);
+
+  AT_DISPATCH_FLOATING_TYPES(self.scalar_type(), "logcumsumexp_out_cpu", [&] {
```

Comment:
can we also add this for complex dtypes?
`AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES`

Reply:
ahh I guess we can't extend this to complex if we are using `max` and `min` ops in the implementation